File: ruby/replace_with_alphabet_position.rb
=begin

PROBLEM:

INPUT(S):

OUTPUTS(S):

ALGORITHM:

=end

ALPHABET = ('a'..'z').to_a

def alphabet_position(text)
  num_str = text.chars.each_with_object('') do |c, str|
    if ALPHABET.include?(c.downcase)
      str << (ALPHABET.index(c.downcase) + 1).to_s + ' '
    end
  end
  num_str.strip
end

# test cases

puts 

File: javascript/replace_with_alphabet_position.js
/*

input:
  - string

output:
  - string with alphabet positions for all alphabetical characters

rules:
  - if a character isn't a letter, ignore it
  - a is 1
  - case insensitive

algorithm:
  - create a constant with alphabet in array (first item empty)
  - initialize a return string to an empty string
  - iterate through the string
    - if letter, put number in return string with a trailing space
  - trim the last space

*/

let alphabetPosition = function (text) {
  const ALPHABET = [ , 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k',
    'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z' ];
  let result = '';
  for (let index = 0; index < text.length; index += 1) {
    let currentCharacter = text[index].toLowerCase();
    if (/[a-z]/.test(currentCharacter)) {
      result += `${ALPHABET.indexOf(currentCharacter)} `;
    }
  }

  return result.trim();
}

console.log(alphabetPosition("The sunset sets at twelve o' clock."));
console.log(alphabetPosition("The narwhal bacons at midnight."));
